### PR TITLE
fix: Reduce the number of requests made on load

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

We currently mount a bunch of logics permanently in order to reduce over-fetching. This has the negative consequence of firing a lot of api requests on page load, regardless of whether they are needed or not.

## Changes

* Changes this so that we "permanently mount" certain logics
* Stops api keys from always being loaded.

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: frontend/src/lib/utils.tsx [hunk 1]
```diff
@@ -1827,3 +1828,13 @@ export function shouldCancelQuery(error: any): boolean {
...
+/**
+ * Some kea logics are used heavily across multiple areas so we keep it mounted once loaded with this trick.
+ */
+export function permanentlyMountLogic(logic: LogicWrapper): void {
+    const mountedLogic = logic.findMounted()
+    if (mountedLogic && !mountedLogic.cache._permanentMount) {
+        mountedLogic.cache._permanentMount = true
+        logic.mount()
+    }
+}
```
Comment (mariusandra) [inline frontend/src/lib/utils.tsx:1840]: You could take the abstraction one step further, and build something like this (pseudocode, didn't run)

```ts
function permanentlyMount(builtLogic: BuiltLogic): void {
    afterMount(() => {
        if (!builtLogic.cache._permanentMount) {
            builtLogic.cache._permanentMount = true
            builtLogic.wrapper.mount()
        }
    )(logic)
}
```

Then you can just add:

```ts
const actionsModel = kea([
    actions(...),
    reducers(...),
    permanentlyMount,
    // other stuff
])
```

... without needing to ask the user to put it into an `afterMount` effect themselves.

See here for docs on "[logic builders](https://keajs.org/docs/meta/kea#logic-builders)"

**Review by mariusandra — APPROVED** [review]
Works and LGTM, though one suggestion for even more abstractions inline

File: frontend/src/lib/utils.tsx (hunk 1 above)
Comment (benjackwhite) [reply frontend/src/lib/utils.tsx:1840]: Ah thats really nice. didnt even think about that 👍

File: frontend/src/lib/utils/kea-logic-builders.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { BuiltLogic, afterMount } from 'kea'
+/**
+ * Some kea logics are used heavily across multiple areas so we keep it mounted once loaded with this trick.
+ */
+export function permanentlyMount(afterMountFn?: (logic: BuiltLogic) => void): (logic: BuiltLogic) => void {
+    return (logic) => {
+        afterMount(() => {
```
Comment (Twixes) [inline frontend/src/lib/utils/kea-logic-builders.ts:7]: I think it'd be slightly clearer the way Marius proposed (keeping `afterMount()` and `permanentlyMount` separate), but that's very minor

File: frontend/src/scenes/App.tsx [hunk 3]
```diff
@@ -88,7 +85,6 @@ export function App(): JSX.Element | null {
     if (showApp) {
         return (
             <>
-                {user && currentTeamId ? <Models /> : null}
```
Comment (Twixes) [inline frontend/src/scenes/App.tsx:91]: ❤️

**Review by Twixes — APPROVED** [review]

File: frontend/src/lib/utils/kea-logic-builders.ts (hunk 2 above)
Comment (benjackwhite) [reply frontend/src/lib/utils/kea-logic-builders.ts:7]: I was mostly thinking of reducing boilerplate but happy to change it

**Pull request merged**